test/mailbox/model/mailbox_tree_builder_test.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+import 'package:model/mailbox/presentation_mailbox.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_node.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_tree.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_tree_builder.dart';
+
+void main() {
+  group('mailbox tree builder test', () {
+    final listPresentationMailbox1 = [
+      PresentationMailbox(MailboxId(Id("1")), parentId: null),
+      PresentationMailbox(MailboxId(Id("2")), parentId: null),
+      PresentationMailbox(MailboxId(Id("3")), parentId: null),
+      PresentationMailbox(MailboxId(Id("4")), parentId: null),
+      PresentationMailbox(MailboxId(Id("11")), parentId: MailboxId(Id('1'))),
+      PresentationMailbox(MailboxId(Id("12")), parentId: MailboxId(Id('1'))),
+      PresentationMailbox(MailboxId(Id("13")), parentId: MailboxId(Id('1')))
+    ];
```

Comment:
- what happen when the input mailbox list have other ordered?
- what happen when the `parentNode` is response after `childNode`?
- what happen when the `grandNode` is response after `childNode` but before `parentNode`?
- what happen when the `parentNode` is response after some of `childNode`?
- what happen when the `grandNode` is response after some of `childNode` but before `parentNode`?
- what happen when the `parentNode` is after some `childNode` and before some `childNode`?